### PR TITLE
github: add Code of Conduct

### PR DESCRIPTION
**Pull request by miri64**
This PR provides a first draft for our future code of conduct. I think it's about high time we get one (though I also am convinced that we already followed this rules to our best abilities for the most part up to this) ;-).

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+allowing everyone to participate.
+
+This code of conduct applies to all spaces managed by the RIOT community. This
+includes IRC, the mailing lists, our GitHub projects, face to face events, and
+any other forums created by the community for communication within the
+community. In additions, violations of this code outside these spaces may also
+affect a person's ability to participate within them.
+
+If you believe someone is violating the code of conduct, we ask that you report
+it by emailing <riot@riot-os.org>. For more details please see our [Reporting
```
Comment (miri64) [inline CODE_OF_CONDUCT.md]: Might be a good idea to have a separate mailing list (or at least an alias to one of the existing ones) for that.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+RIOT-OS Code of Conduct
+=======================
+
+RIOT, like so many other free and open source software projects, is made up of a
+mixed group of professionals and volunteers from all over the world, working on
+every aspect of our goals - including teaching, steering, and connecting people.
+
+We want diversity to be one of our greatest strength, but it can also lead to
```
Comment (kaspar030) [inline CODE_OF_CONDUCT.md]: s/strength/strengths/

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+This is not meant to be an exhaustive list of things you are not allowed to do.
+We rather would like you to think of it as a guide to enrich our community and
+the technical community in general with new knowledge and perspectives by
+allowing everyone to participate.
+
+This code of conduct applies to all spaces managed by the RIOT community. This
+includes IRC, the mailing lists, our GitHub projects, face to face events, and
+any other forums created by the community for communication within the
+community. In additions, violations of this code outside these spaces may also
```
Comment (kaspar030) [inline CODE_OF_CONDUCT.md]: addition

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+it by emailing <riot@riot-os.org>. For more details please see our [Reporting
+Guidelines](TODO).
+
+-   **Be friendly and patient.**
+-   **Be welcoming.** We strive to be a community that welcomes and supports
+    people of all backgrounds and identities. This includes, but is not limited
+    to members of any race, ethnicity, culture, national origin, colour,
+    immigration status, social and economic class, educational level, sex,
+    sexual orientation, gender identity and expression, age, size, family
+    status, political belief, religion, and mental and physical ability.
```
Comment (kaspar030) [inline CODE_OF_CONDUCT.md]: should we scratch political belief?

**Comment by aabadie** [comment]
I was also looking at the Django CoC which is itself inspired from [there](http://web.archive.org/web/20141109123859/http://speakup.io/coc.html).

Thanks for providing this !

**Comment by miri64** [comment]
> I was also looking at the Django CoC which is itself inspired from there.

I know, but since I took the Django CoC as a basis and tweaked it a little for our community I was referencing it insteado of the Speak Up! CoC ;-).

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    err and blaming each other doesn’t get us anywhere. Instead, focus on
+    helping to resolve issues and learning from mistakes.
+
+Text based on the Code of Conduct of the [Django
+community](https://www.djangoproject.com/conduct/).
+
+Questions?
+----------
+
+If you have questions, please see [the FAQ](TODO). If that doesn't answer your
```
Comment (aabadie) [inline CODE_OF_CONDUCT.md]: And we also need a FAQ now :)

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (miri64) [reply CODE_OF_CONDUCT.md]: No, I think it is as important to be in here as e.g. religion. Political beliefs that are not in line with the code of conduct are implicitly exempt anyway, as I read it. ;-)

File: CODE_OF_CONDUCT.md (hunk 5 above)
Comment (miri64) [reply CODE_OF_CONDUCT.md]: We [have one](https://github.com/RIOT-OS/RIOT/wiki/FAQ), but this one (at least the linked one by Django) is supposed to be CoC specific.

**Comment by smlng** [comment]
@miri64 nice, and thanks for tackling this! I like the text and the adaption made for RIOT. I wouldn't fiddle to much with it, but rather have it in place i.e. published. 

How do you/we want to proceed with the FAQ? IMHO the Django FAQ fit for RIOT, too - though we don't have a foundation or other form of official organisation (yet). But where to put them: wiki or webpage (i.e., riot-os.org)?

**Comment by miri64** [comment]
> But where to put them: wiki or webpage (i.e., riot-os.org)?

I'd say the wiki.

**Comment by miri64** [comment]
Reporting Guidelines need also be created.

**Comment by miri64** [comment]
Since this is something involving the whole community, I'd like to have some more input before we go ahead and merge this.

**Comment by gebart** [comment]
I think this is a great addition to the project. 👍

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (kaspar030) [reply CODE_OF_CONDUCT.md]: ok :)

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+allowing everyone to participate.
+
+This code of conduct applies to all spaces managed by the RIOT community. This
+includes IRC, the mailing lists, our GitHub projects, face to face events, and
+any other forums created by the community for communication within the
+community. In addition, violations of this code outside these spaces may also
+affect a person's ability to participate within them.
+
+If you believe someone is violating the code of conduct, we ask that you report
+it by emailing <riot@riot-os.org>. For more details please see our [Reporting
```
Comment (miri64) [inline CODE_OF_CONDUCT.md]: @OlegHahm can we make an alias or something, like conduct@riot-os.org or coc@riot-os.org?

**Comment by miri64** [comment]
Please also discuss the wiki pages I added right now

* [CoC: Reporting Guidelines](https://github.com/RIOT-OS/RIOT/wiki/CoC:-Reporting-Guidelines)
* [CoC: FAQ](https://github.com/RIOT-OS/RIOT/wiki/CoC:-FAQ)

They were adapted from the respective versions by Django as well.

One thing open is: the Django community has the DSF who has a board and also working group that explicitly deals with CoC reports. The latter I replaced with our core maintainers (basically everyone who receives mails to `riot@riot-os.org`), but the first is a little trickier (I marked this with TODO on the page for now). The board basically steps in, if someone is not happy about the decision the CoC-WG made. My initial idea is to maybe add some contact information well trusted members of the RIOT community there, who dealt with such administrative tasks in the past, but maybe someone else has a better idea that we can use before the RIOT association is official ;-).

**Comment by amcjen** [comment]
I'm super pleased to see this effort taken seriously and to use existing code language from other successful projects.  I don't have any suggestions, other than the mention to make it clear how to report if something is amiss--but @miri64 already added them to the Wiki pages, so perfect.  👍

**Comment by smlng** [comment]
> Please also discuss the wiki pages I added right now
>
> CoC: Reporting Guidelines
> CoC: FAQ
> They were adapted from the respective versions by Django as well.

maybe add a footnote, that we adapted the FAQ from Django?

**Comment by miri64** [comment]
> maybe add a footnote, that we adapted the FAQ from Django?

Done, and also for the reporting guidelines.

**Comment by smlng** [comment]
4 approvals in 1 week, looks good - lets see if Murdock agrees too 😉

**Comment by smlng** [comment]
well ... squash please.

File: CODE_OF_CONDUCT.md (hunk 6 above)
Comment (kaspar030) [reply CODE_OF_CONDUCT.md]: Not sure I like the sound of the latter in combination with this context.

Comment (smlng) [reply CODE_OF_CONDUCT.md]: in other contexts sometimes `abuse@example.com` is used, otherwise `conduct@riot-os.org` sounds fine, too?!

**Comment by miri64** [comment]
> 4 approvals in 1 week, looks good - lets see if Murdock agrees too :wink:

Well there is still the issue of accountability we need to discuss: Where should people go, if the core team fails to address a report properly in their opinion about CoC violation? Or should we remove this section for now and reinstate this with the board of the association as soon as it is registered?

**Comment by miri64** [comment]
(the section I'm talking about is the last paragraph of the reporting guidelines)

**Comment by smlng** [comment]
mhm, so this reconsideration is referring to another _escalation stage_ in the reporting process, right? So if a reporting party does not agree with the initial decision/result of the core community, they might take it to the next step. Which in itself could also fail to agree with the expected outcome of the reporter - so there might always be those cases which are difficult to decide or act upon or come to final conclusion everybody happy with. Further as there is no higher board or association for RIOT (yet) its IMHO fine to remove that paragraph for now - and reconsider as soon as that changes.

**Comment by miri64** [comment]
Yes it is basically about what you would call in legal terms a "second instance". The "third instance" (if that fails) is in case of the RIOT community to leave the community (which depending on the case could be very bad for the community, so the pressure to make the "right" decission for each instance is higher the less instances there are).

File: CODE_OF_CONDUCT.md (hunk 6 above)
Comment (miri64) [reply CODE_OF_CONDUCT.md]: Oh yes, this is another issue, that needs to be addressed before merging this IMHO.

**Comment by miri64** [comment]
Ok two issues that are open still:

* What would be our second instance for reporting (or remove the mention of that for now)?
* I can haz abuse@riot-os.org, conduct@riot-os.org or something of the like.

Finally, due to the impact of this on the community I would like to keep this open even with 5 ACKs at least until end of next week.

File: CODE_OF_CONDUCT.md (hunk 6 above)
Comment (miri64) [reply CODE_OF_CONDUCT.md]: > Not sure I like the sound of the latter in combination with this context.

Ooops, only notice this now ^^"

**Comment by miri64** [comment]
I noticed, that I can create the alias `conduct@riot-os.org` myself and did so. Will adapt the text(s)

**Comment by miri64** [comment]
I removed the section on reconsideration in the reporting guidelines for now. Let's just hope that in the meantime it will never come to that case.

**Comment by miri64** [comment]
Squashed and ready to be merged.

**Comment by smlng** [comment]
Let's do this 🎉

**Pull request merged**